var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -462,6 +463,10 @@ def configure_args(self):
         if 'schedulers=auto' not in spec:
             config_args.extend(self.with_or_without('schedulers'))
 
+        if not spec.variants['gpfs'].value:
```

Comment:
```suggestion
        if '~gpfs' in spec:
```

Reply:
I hadn't done it this way because I don't know enough about the definition of `spec` to know whether `if '~gpfs' in spec` is trustworthy; e.g., whether it would still catch if someone specified `gpfs=False` as their variant.

If this is preferred, though, I'll update the PR.

Reply:
I've updated the PR with the proposed change.